circle.yml

```diff
@@ -17,8 +17,10 @@ test:
     - diff -u <(echo -n) <(gofmt -d .)
     - go tool vet *.go                                                   # Go package in root directory.
     - for d in */; do echo $d; done | grep -v tests/ | grep -v third_party/ | xargs go tool vet # All subdirectories except "tests", "third_party".
+    - diff -u <(echo -n) <(go list ./compiler/natives/src/...) # All those packages should have // +build js.
+    - gopherjs install -v net/http # Should build successfully (can't run tests, since only client is supported).
     - >
-      gopherjs test --short --minify
+      gopherjs test -v --short
```

Comment:
Why drop the minify here? 

It actually helps to catch errors in the compiled Javascript (if changes are being made to the GopherJS compiler).

Reply:
Explained in b827aaee653d5db1b538134a3d02433652a10b42 and discussed with @neelance at https://github.com/gopherjs/gopherjs/pull/651#discussion_r128371801, with the follow up done in 2587a1cd5553d8890d4f33cc76fe5eea82ebd7cc.

What are your thoughts after seeing that?

Reply:
@neelance makes the same point about the `--minify` flag essentially being a means of ensuring the generated Javascript code is valid in https://github.com/gopherjs/gopherjs/pull/651#discussion_r128371801

By only testing `--minify` with `fmt` we are assuming that we get complete coverage of generated output... I'm not sure I could easily reason that that's the case.

I take you point about the stack traces... if there's no particular time constraints we could do two test runs, one without `--minify` followed by one with. That way we get the best of both worlds.

Thoughts?

Reply:
In b827aaee653d5db1b538134a3d02433652a10b42, I wrote:

> I don't see a strong reason to use minification during tests. Output
size is not a concern.

You said:

> By only testing `--minify` with `fmt` we are assuming that we get complete coverage of generated output... I'm not sure I could easily reason that that's the case.

I agree. I think that's a good reason to keep `--minify` flag.

It is, after all, the de-facto standard of how majority of production GopherJS code is generated, so testing it is more important than non-minified code. It's not worth sacrificing minified code test coverage that for the sake of convenience of having nicer CI error reports in the (rare) case of errors.

I'll revert that change. Thanks.

> I take you point about the stack traces... if there's no particular time constraints we could do two test runs, one without `--minify` followed by one with. That way we get the best of both worlds.

I don't quite like it because it'd mean you'd see CI build results slower. It's already quite slow at x minutes. Also, it'd feels like a waste of resources, given that the expected behavior is that `--minify` does not change behavior, and I would like that to be testable without running _all_ tests.

Let's leave this for later.

Reply:
> I don't quite like it because it'd mean you'd see CI build results slower.

It ought to be easy to run the non-minified tests only in the case of a failure of the minified tests, to pay that penalty only in the rare case of failure.

Reply:
> It ought to be easy to run the non-minified tests only in the case of a failure of the minified tests, to pay that penalty only in the rare case of failure.

I was thinking that could be close to ideal behavior. But it doesn't seem very easy to me. If you think it's easy, it'd be helpful if you could open an issue and provide details of how to accomplish it.

Reply:
@shurcooL we can easily speed up the tests; they're effectively bound to a single core the way we run them. I've got some rather hacky code locally to runs `gopherjs test X` concurrently up to the number of cores. I'll try and polish that up and submit a PR. It works in much the same way as the runner in `tests/run.go`.